golangci-lint typecheck fails on Windows: Unix-specific syscall symbols in telemetry test (replaces #530)

## Summary
`make check-full` fails on Windows during the `lint` step because `golangci-lint` typechecks `*_test.go` files, and `internal/infrastructure/telemetry/infra_telemetry_extended_test.go` unconditionally imports `syscall` and uses Unix-only symbols.

## Environment
- **OS**: Windows (any version)
- **Go version**: current module version
- **Tool**: `golangci-lint` (via `make lint` → `make check-full`)

## Root Cause
`infra_telemetry_extended_test.go` unconditionally imports `syscall` and uses these platform-specific symbols:

| Symbol | Function | Line(s) |
|--------|----------|---------|
| `syscall.Mkfifo` | `TestLogTrace_WriteError` | 502 |
| `syscall.Rlimit` | `TestLogTrace_CloseError` | 574 |
| `syscall.Getrlimit` | `TestLogTrace_CloseError` | 575 |
| `syscall.RLIMIT_FSIZE` | `TestLogTrace_CloseError` | 575, 578, 582 |
| `syscall.Setrlimit` | `TestLogTrace_CloseError` | 578, 582 |
| `syscall.Umask` | `TestOpenLogFileForAppend_MkdirAllSucceedsButRetryFails` | 784–785 |

These symbols are defined in `syscall` on Unix but **do not exist** in the Windows `syscall` package.

### Why `go build` passes but lint fails
- `go build` only compiles non-test `*.go` files → test file ignored → build succeeds
- `golangci-lint` typechecks **all** Go files including `*_test.go` → typechecker sees undefined symbols → lint fails with exit code 7

## Error Output
```
level=error msg="[linters_context] typechecking error"
internal/infrastructure/telemetry/infra_telemetry_extended_test.go:502:20: undefined: syscall.Mkfifo
internal/infrastructure/telemetry/infra_telemetry_extended_test.go:574:19: undefined: syscall.Rlimit
internal/infrastructure/telemetry/infra_telemetry_extended_test.go:575:20: undefined: syscall.Getrlimit
internal/infrastructure/telemetry/infra_telemetry_extended_test.go:575:38: undefined: syscall.RLIMIT_FSIZE
internal/infrastructure/telemetry/infra_telemetry_extended_test.go:578:29: undefined: syscall.Setrlimit
internal/infrastructure/telemetry/infra_telemetry_extended_test.go:578:47: undefined: syscall.RLIMIT_FSIZE
internal/infrastructure/telemetry/infra_telemetry_extended_test.go:582:20: undefined: syscall.Setrlimit
internal/infrastructure/telemetry/infra_telemetry_extended_test.go:582:38: undefined: syscall.RLIMIT_FSIZE
internal/infrastructure/telemetry/infra_telemetry_extended_test.go:784:22: undefined: syscall.Umask
```

## Fix: Follow the established `platform_dirs_*_test.go` precedent

This project already has a proven pattern for platform-specific test splitting, established in commit `3212c97` and governed by **ADR-020** (`docs/adr/2026-04-windows-compatibility-strategy.md`). The fix should follow that exact convention:

### Precedent (security package)
```
platform_dirs_test.go          ← platform-independent (shared helpers + 5 exemption tests)
platform_dirs_unix_test.go     ← //go:build !windows  (7 unix-only tests)
platform_dirs_windows_test.go  ← //go:build windows   (9 windows-only tests)
```

### Apply to telemetry package

1. **`infra_telemetry_extended_test.go`**: Remove the `syscall` import. Keep only platform-independent tests:
   - `TestSessionCostTracker_Extended`
   - `TestRegisterMetrics_Extended`
   - `TestRecordSessionCost_Extended`
   - `TestTraceTelemetry` (non-FIFO subtests)
   - `TestLedger_Extended`
   - `TestMetricsManager_LoadHistory_Corrupted`
   - `TestMetricsManager_Retention`
   - `TestMetricsManager_LoadRetentionDays`
   - `TestResolveUsageForSummary_NoTracker`
   - `TestIsStale_NonExistent`
   - `TestLogTrace_EmptyTraceFile`
   - `TestLogTrace_NilTrace`
   - `TestLogTrace_CancelledContext`
   - `TestLogTrace_OpenError`
   - `TestLogTrace_MarshalErrorUnreachable`
   - `TestOpenLogFileForAppend_Success`
   - `TestOpenLogFileForAppend_MkdirAllSucceeds`
   - `TestOpenLogFileForAppend_MkdirAllFails`
   - `TestOpenLogFileForAppend_PermissionDenied`

2. **`infra_telemetry_extended_unix_test.go`** (new, with `//go:build !windows`): Contains the three Unix-only tests that use `syscall`:
   - `TestLogTrace_WriteError` (uses `syscall.Mkfifo`)
   - `TestLogTrace_CloseError` (uses `syscall.Rlimit`, `Getrlimit`, `Setrlimit`, `RLIMIT_FSIZE`)
   - `TestOpenLogFileForAppend_MkdirAllSucceedsButRetryFails` (uses `syscall.Umask`)

3. **No `_windows_test.go` stubs needed** — the build-tag exclusion is sufficient documentation (matching the precedent where `platform_dirs_unix_test.go` has no corresponding stub for each Unix-only test).

### Key convention choices
| Choice | Rationale |
|--------|-----------|
| `//go:build !windows` (not `linux \|\| darwin`) | Matches `platform_dirs_unix_test.go` exactly; avoids enumerating every Unix-like platform |
| `_unix_test.go` suffix | Matches `platform_dirs_unix_test.go` naming convention |
| No `t.Skip` stubs on Windows | Matches precedent: `platform_dirs_unix_test.go` has 7 tests with no corresponding stubs |

## Impact
- **Windows CI/CD**: Currently blocked — `make check-full` cannot pass
- **Cross-platform development**: Windows contributors cannot run the full quality pipeline locally
- **ADR compliance**: Fix aligns with ADR-020's established multi-layered Windows compatibility strategy
- **No runtime impact**: Production build is unaffected

## References
- **ADR-020**: `docs/adr/2026-04-windows-compatibility-strategy.md`
- **Precedent commit**: `3212c97` — split `TestIsSystemDirectory_PlatformAware` into `platform_dirs_*_test.go`
- **Existing telemetry platform files**: `system_metrics_linux.go` (`//go:build linux`), `system_metrics_windows.go` (`//go:build windows`)
- **Supersedes**: #530

--- a/internal/infrastructure/telemetry/infra_telemetry_extended_test.go
+++ b/internal/infrastructure/telemetry/infra_telemetry_extended_test.go
@@ -10,7 +10,6 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
-	"syscall"
 	"testing"
 	"time"
 
@@ -495,104 +494,6 @@ func TestLogTrace_OpenError(t *testing.T) {
 	}
 }
 
-func TestLogTrace_WriteError(t *testing.T) {
-	// NOT parallel — uses FIFO coordination that is timing-sensitive.
-	fifoPath := filepath.Join(t.TempDir(), "fifo")
-
-	if err := syscall.Mkfifo(fifoPath, 0666); err != nil {
-		t.Skipf("cannot create fifo (non-Unix?): %v", err)
-	}
-
-	trace := &domain_telemetry.TurnTrace{FinalStatus: "fifo-test"}
-
-	// Strategy: reader opens (unblocks writer's open), then immediately closes.
-	// When logTrace then calls Write, the reader is gone → EPIPE.
-	readerOpened := make(chan struct{})
-	readerClosed := make(chan struct{})
-
-	go func() {
-		r, err := os.OpenFile(fifoPath, os.O_RDONLY, 0)
-		if err != nil {
-			close(readerOpened)
-			return
-		}
-		close(readerOpened) // signal: reader connected (both opens completed)
-		_ = r.Close()       // immediately close — writer gets EPIPE on next write
-		close(readerClosed)
-	}()
-
-	logDone := make(chan struct{})
-	go func() {
-		logTrace(context.Background(), fifoPath, trace)
-		close(logDone)
-	}()
-
-	// Wait for reader to connect (unblocks logTrace's open).
-	<-readerOpened
-	// Wait for reader to fully close before logTrace marshals and writes.
-	<-readerClosed
-	// Now logTrace will marshal and write — reader is gone, so f.Write gets EPIPE.
-	<-logDone
-	// If we reach here, the write-error path was exercised (log.Printf warning).
-}
-
-func TestLogTrace_CloseError(t *testing.T) {
-	// NOT parallel — uses RLIMIT_FSIZE to trigger EFBIG on write, and
-	// attempts to also trigger a close(2) error.
-	//
-	// Strategy:
-	//  1. Set RLIMIT_FSIZE to a small value (1KB).
-	//  2. Create a large TurnTrace whose JSON exceeds the limit.
-	//  3. logTrace opens (succeeds), writes → EFBIG (write-error path).
-	//  4. close(2) after a failed O_APPEND write: on Linux/ext4 this
-	//     typically returns 0, so the close-error body is unreachable
-	//     in practice. We retain this test to prove the path exists and
-	//     document the limitation.
-
-	tempDir := t.TempDir()
-	traceFile := filepath.Join(tempDir, "close_err.jsonl")
-
-	// Build a trace whose JSON exceeds ~1KB.
-	execs := make([]domain_telemetry.ToolExecutionTrace, 200)
-	for i := range execs {
-		execs[i] = domain_telemetry.ToolExecutionTrace{
-			ToolName:  "very-long-tool-name-to-fill-bytes",
-			StartTime: time.Now(),
-			Duration:  time.Second,
-			Status:    "success",
-			Error:     "a detailed error message that takes up space in the JSON output",
-		}
-	}
-	trace := &domain_telemetry.TurnTrace{
-		StartTime:         time.Now(),
-		EndTime:           time.Now().Add(time.Minute),
-		InferenceDuration: 30 * time.Second,
-		ToolExecutions:    execs,
-		FinalStatus:       "complete",
-	}
-
-	var orig syscall.Rlimit
-	if err := syscall.Getrlimit(syscall.RLIMIT_FSIZE, &orig); err != nil {
-		t.Skipf("cannot get rlimit: %v", err)
-	}
-	defer func() { _ = syscall.Setrlimit(syscall.RLIMIT_FSIZE, &orig) }()
-
-	lim := orig
-	lim.Cur = 1024 // 1KB — trace JSON is ~30KB
-	if err := syscall.Setrlimit(syscall.RLIMIT_FSIZE, &lim); err != nil {
-		t.Skipf("cannot set rlimit: %v", err)
-	}
-
-	// logTrace will open, marshal, write → EFBIG, then close.
-	// Write-error path is covered. Close-error path depends on kernel/fs.
-	logTrace(context.Background(), traceFile, trace)
-
-	// Verify the file was created (it should exist but be truncated).
-	if fi, err := os.Stat(traceFile); err == nil {
-		t.Logf("file size after EFBIG write: %d bytes", fi.Size())
-	}
-}
-
 // TestLogTrace_MarshalErrorUnreachable documents that the json.Marshal error
 // path in logTrace is unreachable because TurnTrace contains only JSON-
 // serializable fields (sync.Mutex excluded via json:"-"). No test is needed.
@@ -763,40 +664,6 @@ func TestOpenLogFileForAppend_PermissionDenied(t *testing.T) {
 	// Should NOT contain the mkdir-fallback message.
 	if strings.Contains(err.Error(), "also failed to create dir") {
 		t.Error("error should NOT mention mkdir fallback for non-NotExist errors")
-	}
-}
-
-// TestOpenLogFileForAppend_MkdirAllSucceedsButRetryFails covers the branch
-// where the first OpenFile fails with ENOENT, MkdirAll succeeds, but the
-// retry OpenFile still fails. We use syscall.Umask(0o777) so that MkdirAll
-// creates the parent directory with mode 0000, causing the retry to fail
-// with EACCES.
-//
-// NOT parallel — syscall.Umask is process-global.
-func TestOpenLogFileForAppend_MkdirAllSucceedsButRetryFails(t *testing.T) {
-	// NOT parallel: umask is process-global.
-	tmpDir := t.TempDir()
-
-	// Only ONE level of missing directory so MkdirAll itself succeeds.
-	logPath := filepath.Join(tmpDir, "missing", "log.jsonl")
-
-	// Set umask so MkdirAll creates the parent dir with mode 0000.
-	oldUmask := syscall.Umask(0o777)
-	t.Cleanup(func() { syscall.Umask(oldUmask) })
-
-	_, err := openLogFileForAppend(logPath)
-	if err == nil {
-		t.Fatal("expected error, got nil")
-	}
-
-	// Verify the error is from the retry (after mkdir), not from mkdir itself.
-	errStr := err.Error()
-	if !strings.Contains(errStr, "after mkdir") {
-		t.Errorf("error should mention 'after mkdir', got: %v", err)
-	}
-	// Verify the directory WAS created (MkdirAll succeeded).
-	if _, statErr := os.Stat(filepath.Dir(logPath)); os.IsNotExist(statErr) {
-		t.Error("parent directory was not created by MkdirAll")
 	}
 }
 

--- a/internal/infrastructure/telemetry/infra_telemetry_extended_unix_test.go
+++ b/internal/infrastructure/telemetry/infra_telemetry_extended_unix_test.go
@@ -1,0 +1,156 @@
+// Copyright (c) 2026 gosharplite@gmail.com
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package telemetry
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	domain_telemetry "github.com/gosharplite/tell-me-go/internal/domain/telemetry"
+)
+
+// ---------------------------------------------------------------------------
+// Unix-only logTrace / openLogFileForAppend error-path tests.
+// These use syscall symbols (Mkfifo, Rlimit, Getrlimit, Setrlimit, Umask)
+// that are not defined in the Windows syscall package.
+// ---------------------------------------------------------------------------
+
+func TestLogTrace_WriteError(t *testing.T) {
+	// NOT parallel — uses FIFO coordination that is timing-sensitive.
+	fifoPath := filepath.Join(t.TempDir(), "fifo")
+
+	if err := syscall.Mkfifo(fifoPath, 0666); err != nil {
+		t.Skipf("cannot create fifo (non-Unix?): %v", err)
+	}
+
+	trace := &domain_telemetry.TurnTrace{FinalStatus: "fifo-test"}
+
+	// Strategy: reader opens (unblocks writer's open), then immediately closes.
+	// When logTrace then calls Write, the reader is gone → EPIPE.
+	readerOpened := make(chan struct{})
+	readerClosed := make(chan struct{})
+
+	go func() {
+		r, err := os.OpenFile(fifoPath, os.O_RDONLY, 0)
+		if err != nil {
+			close(readerOpened)
+			return
+		}
+		close(readerOpened) // signal: reader connected (both opens completed)
+		_ = r.Close()       // immediately close — writer gets EPIPE on next write
+		close(readerClosed)
+	}()
+
+	logDone := make(chan struct{})
+	go func() {
+		logTrace(context.Background(), fifoPath, trace)
+		close(logDone)
+	}()
+
+	// Wait for reader to connect (unblocks logTrace's open).
+	<-readerOpened
+	// Wait for reader to fully close before logTrace marshals and writes.
+	<-readerClosed
+	// Now logTrace will marshal and write — reader is gone, so f.Write gets EPIPE.
+	<-logDone
+	// If we reach here, the write-error path was exercised (log.Printf warning).
+}
+
+func TestLogTrace_CloseError(t *testing.T) {
+	// NOT parallel — uses RLIMIT_FSIZE to trigger EFBIG on write, and
+	// attempts to also trigger a close(2) error.
+	//
+	// Strategy:
+	//  1. Set RLIMIT_FSIZE to a small value (1KB).
+	//  2. Create a large TurnTrace whose JSON exceeds the limit.
+	//  3. logTrace opens (succeeds), writes → EFBIG (write-error path).
+	//  4. close(2) after a failed O_APPEND write: on Linux/ext4 this
+	//     typically returns 0, so the close-error body is unreachable
+	//     in practice. We retain this test to prove the path exists and
+	//     document the limitation.
+
+	tempDir := t.TempDir()
+	traceFile := filepath.Join(tempDir, "close_err.jsonl")
+
+	// Build a trace whose JSON exceeds ~1KB.
+	execs := make([]domain_telemetry.ToolExecutionTrace, 200)
+	for i := range execs {
+		execs[i] = domain_telemetry.ToolExecutionTrace{
+			ToolName:  "very-long-tool-name-to-fill-bytes",
+			StartTime: time.Now(),
+			Duration:  time.Second,
+			Status:    "success",
+			Error:     "a detailed error message that takes up space in the JSON output",
+		}
+	}
+	trace := &domain_telemetry.TurnTrace{
+		StartTime:         time.Now(),
+		EndTime:           time.Now().Add(time.Minute),
+		InferenceDuration: 30 * time.Second,
+		ToolExecutions:    execs,
+		FinalStatus:       "complete",
+	}
+
+	var orig syscall.Rlimit
+	if err := syscall.Getrlimit(syscall.RLIMIT_FSIZE, &orig); err != nil {
+		t.Skipf("cannot get rlimit: %v", err)
+	}
+	defer func() { _ = syscall.Setrlimit(syscall.RLIMIT_FSIZE, &orig) }()
+
+	lim := orig
+	lim.Cur = 1024 // 1KB — trace JSON is ~30KB
+	if err := syscall.Setrlimit(syscall.RLIMIT_FSIZE, &lim); err != nil {
+		t.Skipf("cannot set rlimit: %v", err)
+	}
+
+	// logTrace will open, marshal, write → EFBIG, then close.
+	// Write-error path is covered. Close-error path depends on kernel/fs.
+	logTrace(context.Background(), traceFile, trace)
+
+	// Verify the file was created (it should exist but be truncated).
+	if fi, err := os.Stat(traceFile); err == nil {
+		t.Logf("file size after EFBIG write: %d bytes", fi.Size())
+	}
+}
+
+// TestOpenLogFileForAppend_MkdirAllSucceedsButRetryFails covers the branch
+// where the first OpenFile fails with ENOENT, MkdirAll succeeds, but the
+// retry OpenFile still fails. We use syscall.Umask(0o777) so that MkdirAll
+// creates the parent directory with mode 0000, causing the retry to fail
+// with EACCES.
+//
+// NOT parallel — syscall.Umask is process-global.
+func TestOpenLogFileForAppend_MkdirAllSucceedsButRetryFails(t *testing.T) {
+	// NOT parallel: umask is process-global.
+	tmpDir := t.TempDir()
+
+	// Only ONE level of missing directory so MkdirAll itself succeeds.
+	logPath := filepath.Join(tmpDir, "missing", "log.jsonl")
+
+	// Set umask so MkdirAll creates the parent dir with mode 0000.
+	oldUmask := syscall.Umask(0o777)
+	t.Cleanup(func() { syscall.Umask(oldUmask) })
+
+	_, err := openLogFileForAppend(logPath)
+	if err == nil {
+		t.Fatal("expected error, got nil")
+	}
+
+	// Verify the error is from the retry (after mkdir), not from mkdir itself.
+	errStr := err.Error()
+	if !strings.Contains(errStr, "after mkdir") {
+		t.Errorf("error should mention 'after mkdir', got: %v", err)
+	}
+	// Verify the directory WAS created (MkdirAll succeeded).
+	if _, statErr := os.Stat(filepath.Dir(logPath)); os.IsNotExist(statErr) {
+		t.Error("parent directory was not created by MkdirAll")
+	}
+}

